sdk/tests/conformance2/rendering/builtin-vert-attribs.html

```diff
@@ -0,0 +1,408 @@
+        gl_Position.x = (prim_vert_id == 1) ? 2.0 : -1.0;
+        gl_Position.y = (prim_vert_id == 2) ? 2.0 : -1.0;
+        v_color = vec4(1.0, float(flat_vert_id) / 255.0, float(${iid.name}) / 255.0, 1.0);
+#if ${(iid.name == 'gl_InstanceID' && ERRATA.IGNORE_GL_INSTANCE_ID)|0}
+        v_color.b = 0.0;
+#endif
+    }`;
+}
+
+const FS = `\
+    #version 300 es
+    precision mediump float;
+
+    in vec4 v_color;
+    out vec4 o_color;
+
+    void main() {
+        o_color = v_color;
+    }
+`;
+
+
+function crossCombine(...args) {
+  function crossCombine2(listA, listB) {
+    const listC = [];
+    for (const a of listA) {
+      for (const b of listB) {
+        const c = Object.assign({}, a, b);
+        listC.push(c);
+      }
+    }
+    return listC;
+  }
+
+  let res = [{}];
+  while (args.length) {
+    const next = args.shift();
+    next[0].defined;
+    res = crossCombine2(res, next);
+  }
+  return res;
+}
+
+/// makeCombiner('foo', [5, 3]) -> [{foo: 5}, {foo: 3}]
+function makeCombiner(key, vals) {
+  const ret = [];
+  for (const val of vals) {
+      const cur = {};
+      cur[key] = val;
+      ret.push(cur);
+  }
+  return ret;
+}
+
+debug('Draw a point with a shader that takes no attributes and verify it fills the whole canvas.');
+
+
+let TESTS = [
+    makeCombiner('vid', [
+        {name: 'a_VertexID', loc:0},
+        {name: 'a_VertexID', loc:2}, // Test 2, so that we're not only testing 0.
+        {name: 'gl_VertexID', loc:-1},
+        {name: 'gl_VertexID', loc:0}, // Enable a vertex array, despite not using it.
+        {name: 'gl_VertexID', loc:2}, // Enable a vertex array, despite not using it.
+    ]),
+    makeCombiner('iid', [
+        {name: 'a_InstanceID', loc:1},
+        {name: 'gl_InstanceID', loc:-1},
+        {name: 'gl_InstanceID', loc:1}, // Enable a vertex array, despite not using it.
+    ]),
+    makeCombiner('separate_vbufs', [true, false]),
+];
+//console.log('a', {TESTS});
+TESTS = crossCombine(...TESTS);
+//console.log('b', {TESTS});
+
+
+let vdata = new Int32Array(1000);
+vdata = vdata.map((v,i) => i);
+const vbuf = gl.createBuffer();
+gl.bindBuffer(gl.ARRAY_BUFFER, vbuf);
+gl.bufferData(gl.ARRAY_BUFFER, vdata, gl.STATIC_DRAW);
+
+
+const vbuf2 = gl.createBuffer();
+gl.bindBuffer(gl.ARRAY_BUFFER, vbuf2);
+gl.bufferData(gl.ARRAY_BUFFER, vdata, gl.STATIC_DRAW);
+
+
+let index_data = new Uint32Array(1000);
+index_data = index_data.map((x,i) => 10+i);
+const index_buffer = gl.createBuffer();
+gl.bindBuffer(gl.ELEMENT_ARRAY_BUFFER, index_buffer);
+gl.bufferData(gl.ELEMENT_ARRAY_BUFFER, index_data, gl.STATIC_DRAW);
+
+
+gl.disable(gl.DEPTH_TEST);
+
+(async () => {
+    for (const desc of TESTS) {
+        await wtu.dispatchPromise(); // Yield, for responsiveness.
+        debug('');
+        debug('---------------------');
+        debug(`desc: ${JSON.stringify(desc)}`);
+
+        let fn = (vs) => {
+            //console.log({vs});
+            const prog = wtu.setupProgram(gl, [vs, FS]);
```

Comment:
Here, looks vs, fs should be ids of DOM elements, instead of source shaders, I think.

Reply:
Oh, sorry, how careless, missed other branches of if